File: Application/Classes/Base/BaseView.m
//
//  BaseView.m
//  NikmeApps
//
//  Created by Linh NGUYEN on 9/6/14.
//  Copyright (c) 2014 Nikmesoft Ltd. All rights reserved.
//

#import "BaseView.h"

@implementation BaseView

- (instancetype)initWithNibName:(NSString*)nibNameOrNil {
    self = [super init];
    NSString *nibName = NSStringFromClass([self class]);
    if ([[NSBundle mainBundle] pathForResource:nibName ofType:@"nib"])
    {
        NSArray *arrayOfViews = [[NSBundle mainBundle] loadNibNamed:nibName owner:self options:nil];
        for (id v in arrayOfViews) {
            if([v isKindOfClass:[self class]])
            {
                self = v;
            }
        }
    }
    return self;
}

- (instancetype)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
    if(self) {
        [self viewDidLoad];
    }
    return self;
}

- (void)awakeFromNib {
    [super awakeFromNib];
    [self viewDidLoad];
}

- (void)viewDidLoad {
    
}

#pragma mark - Show/Hide Method
- (void)showInView:(UIView*)parentView animation:(UIViewAnimationOptions)animation
          duration:(NSTimeInterval)duration {
    [UIView transitionWithView:parentView duration:duration options:animation animations:^{
        [parentView addSubview:self];
    } completion:^(BOOL finished) {
        
    }];
}

- (void)dismissWithAnimation:(UIViewAnimationOptions)animation duration:(NSTimeInterval)duration {
    [UIView animateWithDuration:duration delay:0.0 options:animation
                     animations:^{self.alpha = 0.0;}
                     completion:^(BOOL finished){ [self removeFromSuperview]; }];
}


@end

File: CoAssetsApps/COCore/Classes/Base/BaseView.swift
//
//  BaseView.swift
//  CoAssets-Agent
//
//  Created by Linh NGUYEN on 12/12/15.
//  Copyright © 2015 Nikmesoft Ltd. All rights reserved.
//

import UIKit

class BaseView: UIView {

    override init(frame: CGRect) {
        super.init(frame: frame)
        viewDidLoad()
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        viewDidLoad()
    }

    func viewDidLoad() {
        
    }
    
    class func nibView(nibName: String, bundle: NSBundle?) -> BaseView {
        if let view = UINib(nibName: nibName, bundle: bundle).instantiateWithOwner(nil, options: nil)[0] as? BaseView {
            return view
        }
        return BaseView()
    }
    
    class func nibView() -> BaseView {
        let view = BaseView.nibView(self.className, bundle: nil)
        return view
    }
}

class BaseViewPorfolio: UIView {
    override func layoutSubviews() {
        super.layoutSubviews()
        self.layer.borderWidth = 1.5
        self.layer.borderColor = UIColor.blackColor().CGColor
    }
}
